py3status/modules/external_script.py

```diff
@@ -64,8 +72,9 @@ def external_script(self):
         response = {}
         response['cached_until'] = self.py3.time_in(self.cache_timeout)
         try:
-            output = self.py3.command_output(self.script_path, shell=True, localized=self.localize)
-            output_lines = output.splitlines()
+            self.output = self.py3.command_output(
+                self.script_path, shell=True, localized=self.localize)
+            output_lines = self.output.splitlines()
```

Comment:
Leave this alone as-is. Put `output` inside the dict instead... and assign `self.py3.safe_format()` to `self.composite`. Feed composite to `self.py3.notify_user`.. then we can haz all same placeholders.

Don't document `{output_full}`. Fewer behavior changes/breakages by skipping out on making new configs/placeholders only to be deprecated later, eg `button_show_notification` and hardcoded `self.output`.

You're almost there.

Reply:
I don't want to tackle formatting in this PR, I don't like forever-hanging huge PRs with lots of unrelated changes. This PR is solving one specific problem - showing full output in a notification without refreshing module on click. Formatting needs to be reviewed here anyway: `output` needs to not be truncated, `line_?` needs to be introduced, color parsing needs to be improved, etc - all of this deserves to be done separately.

Reply:
We're not tackling formatting. We're just adding `output_full` to a dict... doing same `self.output =` thing, but in different place... and with composite instead of string, and passing the composite to notify_user.. so we don't have to run same thing twice on every click.

Reply:
I just don't see the benefit of doing this, at this moment it only complicates the code. When there are more formatter placeholders, of course it makes sense to pre-compute them all and store in a dict, and do some micro-optimizations like wrapping in a composite - but not right now when there are so many unresolved issues and formatting itself needs to be redesigned.

Reply:
Just pushed out modifications. I think this does what you want... and it looks like we don't really need `button_refresh`... so we can hide it... and I was slightly off... I don't want you to pass a composite. I wanted you to make and use `self.dict` instead... so all format have same placeholders.

Reply:
I think we do need `button_refresh`, I want the middle click (button `2`) to be a silent refresh without any notification (because the notification it will show will contain outdated info anyway).

Reply:
Sensible. I pushed it out. You can use `format_notification = '{output_full}'`... and the `format` support hidden `{output_full}` too. Things look okay on your side? Squash this PR?